src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -941,10 +941,945 @@ class spell_silverpine_forsaken_trooper_masterscript_high_command : public Spell
+
+    EVENT_START_QUEST_EXSANGUINATE          = 1,
+    EVENT_WAIT_FOR_PLAYER_EXSANGUINATE      = 3,
+    EVENT_RANE_HIDE                         = 4,
+    EVENT_SET_FACE_TO_BLOODFANG             = 5,
+    EVENT_RANE_TALK_TO_PLAYER               = 7,
+    EVENT_RANE_LAST_MOVE                    = 8,
+
+    ACTION_RANE_JUMP_DEATH                  = 1,
+    ACTION_RANE_SKIP_PATH                   = 2,
+
+    TALK_YORICK_EXSANGUINATE_SUMMON         = 0,
+    TALK_YORICK_EXSANGUINATE_HIDE           = 1,
+
+    PATH_YORICK_UP                          = 448820,
+    PATH_YORICK_HIDE                        = 448821,
+
+    WAYPOINT_CLOSE_TO_ARMOIRE               = 15,
+    WAYPOINT_HIDDEN_NEXT_TO_ARMOIRE         = 2
+};
+
+// Deathstalker Rane Yorick - 44882
+struct npc_silverpine_deathstalker_rane_yorick : public ScriptedAI
+{
+    npc_silverpine_deathstalker_rane_yorick(Creature* creature) : ScriptedAI(creature), _playerArrived(false), _playerSkipped(false) { }
+
+    void JustAppeared() override
+    {
+        me->SetPowerType(POWER_ENERGY);
+        me->SetMaxPower(POWER_ENERGY, 100);
+        me->SetPower(POWER_ENERGY, 100, true);
+    }
+
+    void IsSummonedBy(WorldObject* summoner) override
+    {
+        if (Player* player = summoner->ToPlayer())
+            _playerGUID = player->GetGUID();
+
+        _events.ScheduleEvent(EVENT_START_QUEST_EXSANGUINATE, 1s);
+    }
+
+    void WaypointReached(uint32 waypointId, uint32 pathId) override
+    {
+        if (pathId == PATH_YORICK_UP && waypointId == WAYPOINT_CLOSE_TO_ARMOIRE)
+            _events.ScheduleEvent(EVENT_WAIT_FOR_PLAYER_EXSANGUINATE, 1s);
+
+        if (pathId == PATH_YORICK_HIDE && waypointId == WAYPOINT_HIDDEN_NEXT_TO_ARMOIRE)
+        {
+            me->SetFacingTo(4.6425757f);
+
+            DoCastSelf(SPELL_STEALTH);
+
+            _playerSkipped = true;
+        }
+    }
+
+    void DoAction(int32 param) override
+    {
+        switch (param)
+        {
+            case ACTION_RANE_JUMP_DEATH:
+                me->SetDisableGravity(true);
+                _events.ScheduleEvent(EVENT_RANE_LAST_MOVE, 10ms);
+                break;
+
+            case ACTION_RANE_SKIP_PATH:
+            {
+                me->PauseMovement();
+
+                me->GetMotionMaster()->Clear();
+
+                me->NearTeleportTo(YorickReady, false);
+
+                _events.Reset();
+
+                _events.ScheduleEvent(EVENT_SET_FACE_TO_BLOODFANG, 1s);
+
+                DoCastSelf(SPELL_STEALTH);
+                break;
+            }
+
+            default:
+                break;
+        }
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+        {
+            // Note: not sure if this counts as a hackfix as she doesn't seem to inherit the non-cosmetic phase.
+            if (player->GetPhaseShift().HasPhase(PHASE_WAITING_TO_EXSANGUINATE))
```

Comment:
should be done in a generic way (and iirc it is done, at least for personal clones?)

Reply:
This is not a personal clone though, everyone is able to see her. And after checking three hundred times, there's no other spell being cast or serverside spell that approaches the quests' spellIds that would through some information about this. Not exactly sure why she doesn't inherit a non-cosmetic phase change.

Reply:
Removing this piece of code ahead of https://github.com/TrinityCore/TrinityCore/pull/28345 being merged since this fixes the phasing inheritance issue.